app/controllers/api/promotions_controller.rb

```diff
@@ -18,5 +18,25 @@ def show
              locals: { promotion: promotion }
     end
 
+    private
+
+    def companies_aproved
+      @companiesIds = []
+      Employee.where(status: "approved", role: "partner").each do |employee|
+        @companiesIds.push(employee.company_id)
```

Comment:
:wat:

Reply:
Company.joins(:employess).where(employees: {status: approved, role: partner}) algo así.

Reply:
Company.joins(:employees).select("id").where(employees: {status: "approved", role: "partner"}) esta si es la certera. 
La consulta queda algo así:
`SELECT "companies"."id" FROM "companies" INNER JOIN "employees" ON "employees"."company_id" = "companies"."id" WHERE "employees"."status" = $1 AND "employees"."role" = $2  [["status", "approved"], ["role", "partner"]]`